core/src/main/java/com/sequenceiq/cloudbreak/service/RotateSaltPasswordService.java

```diff
@@ -62,14 +77,47 @@ private boolean isChangeSaltuserPasswordSupported(StackDto stack) {
 
     public void rotateSaltPassword(StackDto stack) throws CloudbreakOrchestratorException {
         validateRotateSaltPassword(stack);
-        SecurityConfig securityConfig = securityConfigService.getOneByStackId(stack.getId());
+        SecurityConfig securityConfig = stack.getSecurityConfig();
         String oldPassword = securityConfig.getSaltSecurityConfig().getSaltPassword();
         String newPassword = PasswordUtil.generatePassword();
         List<GatewayConfig> allGatewayConfig = gatewayConfigService.getAllGatewayConfigs(stack);
         hostOrchestrator.changePassword(allGatewayConfig, newPassword, oldPassword);
         securityConfigService.changeSaltPassword(securityConfig, newPassword);
     }
 
+    public void rotateSaltPasswordFallback(StackDto stack) throws CloudbreakOrchestratorFailedException {
+        List<GatewayConfig> allGatewayConfig = gatewayConfigService.getAllGatewayConfigs(stack);
+
+        try {
+            Map<String, String> response = hostOrchestrator.runCommandOnHosts(allGatewayConfig, stack.getAllPrimaryGatewayInstanceNodes(),
+                    SALTUSER_DELETE_COMMAND);
+            LOGGER.debug("Saltuser delete command response: {}", response);
+        } catch (CloudbreakOrchestratorFailedException e) {
+            LOGGER.warn("Failed to run saltuser delete command, assuming it is already deleted", e);
+        }
```

Comment:
could you move this to a method called something like `removeSaltUserFromGateways` or similar